### ISSUE TITLE
Constructor in interface should have return type hint

### ISSUE THREAD

**Issue by Aistina** [scrutinizer-ci/php-analyzer#540]
Scrutinizer reports one minor issue in my interface that requires implementors to have a constructor taking one argument, then when I add it it reports constructors should not have return type hints.

```php
interface ReporterInterface
{
    /**
     * (snip)
     *
     * @param string|null $argument
     */
    public function __construct($argument);
```

see https://scrutinizer-ci.com/g/Aistina/phpta/inspections/1e253d72-e676-495f-9949-1d9cceaa4323/issues/files/src/PHPTA/Processors/ReporterInterface.php?status=new&orderField=path&order=asc&issueId=cb8f300d-a084-4841-87c1-527accfd9dd9

**Comment by neomerx** [comment]
Why do you think it's a bug? Why not just adding to the comment
~~~
@return void
@return mixed
or something more suitable

**Comment by Aistina** [comment]
Because if I do add `@return void`, I get the following error instead:

> Adding a @returnannotation to constructors is generally not recommended
as a constructor does not have a meaningful return value.

Which in my opinion is more correct, since constructors do indeed never
have a return value.

**Comment by neomerx** [comment]
I think constructor can't return void. It returns $this

**Comment by Aistina** [comment]
I don't think so, it's not like a constructor would ever have a `return` statement. Like Scrutinizer itself says:

> [..] as a constructor does not have a meaningful return value.

Hence I think that this error should not be reporter for abstract constructors:

> For interfaces and abstract methods it is generally a good practice to add a `@return` annotation even if it is just `@return void` or `@return null`, so that implementors know what to do in the overridden method.

**Comment by neomerx** [comment]
Constructor can have @return but it will require specifing class and 'void' will cause an error. It might accept $this but I haven't checked.

**Comment by neomerx** [comment]
@return $this doesn't work either

**Comment by schmittjoh** [comment]
Hey guys, just to chime in on this discussion, I think @Aistina has a point that we should remove the message asking for a return comment on a constructor even if it is on an interface. 

The reasoning behind this is simply that whatever you return from this method is going to be discarded by PHP, and is never used anywhere.

**Comment by neomerx** [comment]
Agree

**Comment by luketarplin** [comment]
+1
